src/FM.LiveSwitch.Mux.Standard/Muxer.cs

```diff
@@ -378,37 +393,48 @@ private async Task<LogEntry[]> GetLogEntries(MuxOptions options)
                             filePaths = Directory.EnumerateFiles(_Options.InputPath, "*.*", SearchOption.TopDirectoryOnly);
                         }
 
-                        foreach (var filePath in filePaths)
-                        {
-                            // filter the input files if a filter is provided.
-                            if (_Options.InputFilter != null && !Regex.Match(Path.GetFileName(filePath), _Options.InputFilter).Success)
-                            {
-                                continue;
-                            }
-
-                            if (filePath.EndsWith(".json") || filePath.EndsWith(".json.rec"))
-                            {
-                                try
-                                {
-                                    logEntries.AddRange(await _LogUtility.GetEntries(filePath, _Logger).ConfigureAwait(false));
-                                }
-                                catch (FileNotFoundException)
-                                {
-                                    _Logger.LogWarning("Could not read from {FilePath} as it no longer exists. Is another process running that could have removed it?", filePath);
-                                }
-                                catch (IOException ex) when (ex.Message.Contains("Stale file handle")) // for Linux
-                                {
-                                    _Logger.LogWarning("Could not read from {FilePath} as the file handle is stale. Is another process running that could have removed it?", filePath);
-                                }
-                            }
+                        var validPaths =
+                            from path in filePaths
+                            where _Options.InputFilter == null || Regex.Match(Path.GetFileName(path), _Options.InputFilter).Success
+                            where path.EndsWith(".json") || path.EndsWith(".json.rec")
+                            where !File.Exists($"{path}.fail")
```

Comment:
Checking for the fail file before processing is the part I mentioned in the PR description. If we want to keep the change as small as possible, including only this would fix the specific recording issue we're seeing. But as I mentioned, the way we order events from all sessions at once is problematic if we can't trust the timestamps on the log events.